### PR TITLE
build: simplify Makefile build process

### PR DESCRIPTION
**Pull request by dundargoc**
These are related commit that servers to simplify the neovim build process.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -120,33 +106,33 @@ build/.ran-third-party-cmake::
...
+	$(SINGLE_MAKE) -C src/nvim/testdir NVIM_PRG=$(NVIM_PRG) SCRIPTS= $(MAKEOVERRIDES) $(patsubst src/nvim/testdir/%.vim,%,$@)
 
 build/runtime/doc/tags helptags: | nvim
-	+$(BUILD_TOOL) -C build runtime/doc/tags
+	$(BUILD_TOOL) -C build runtime/doc/tags
 
 # Builds help HTML _and_ checks for invalid help tags.
-helphtml: | nvim build/runtime/doc/tags
-	+$(BUILD_TOOL) -C build doc_html
+helphtml: nvim build/runtime/doc/tags
```
Comment (kylo252) [inline Makefile:126]:
```suggestion
helphtml:
```

File: Makefile [hunk 2]
```diff
@@ -120,33 +106,33 @@ build/.ran-third-party-cmake::
...
+functionaltest: nvim
+	$(BUILD_TOOL) -C build functionaltest
 
-functionaltest-lua: | nvim
-	+$(BUILD_TOOL) -C build functionaltest-lua
+functionaltest-lua: nvim
+	$(BUILD_TOOL) -C build functionaltest-lua
 
-lualint: | build/.ran-cmake deps
+lualint: build/.ran-cmake deps
```
Comment (kylo252) [inline Makefile:135]:
```suggestion
lualint:
```

File: Makefile [hunk 3]
```diff
@@ -226,4 +212,4 @@ $(DEPS_BUILD_DIR)/%: phony_force
 	$(BUILD_TOOL) -C $(DEPS_BUILD_DIR) $(patsubst $(DEPS_BUILD_DIR)/%,%,$@)
 endif
 
-.PHONY: test lualint pylint shlint functionaltest unittest lint clint clean distclean nvim libnvim cmake deps install appimage checkprefix commitlint
+.PHONY: test lualint pylint shlint functionaltest unittest lint clint clean distclean nvim libnvim cmake deps install appimage checkprefix commitlint oldtest helptags helphtml benchmark clint-full functionaltest-lua generated-sources all check-single-includes
```
Comment (kylo252) [inline Makefile:215]: Might be useful to split this and declare a new `.PHONY` after a bunch of targets.

File: Makefile [hunk 4]
```diff
@@ -45,21 +42,12 @@ ifneq (1,$(words [$(DEPS_BUILD_DIR)]))
...
-      BUILD_TOOL = ninja
-    else
-      # User's version of CMake doesn't support Ninja
-      BUILD_TOOL = $(MAKE)
-      CMAKE_GENERATOR := Unix Makefiles
-    endif
-  else
-    BUILD_TOOL = $(MAKE)
-  endif
+ifeq (Ninja,$(CMAKE_GENERATOR))
```
[the remark this answers is not in the document]
Comment (dundargoc) [reply Makefile:45]: Do you remember why? Do you have a link to the comment?

[the remark this answers is not in the document]
Comment (dundargoc) [reply Makefile:45]: Never mind, found it. Thanks.

File: Makefile [hunk 5]
```diff
@@ -45,21 +42,12 @@ ifneq (1,$(words [$(DEPS_BUILD_DIR)]))
...
+  BUILD_TOOL = ninja
```
[the remark this answers is not in the document]
Comment (dundargoc) [reply Makefile]: ?

**Comment by dundargoc** [comment]
@kylo252 I didn't realize Justin was against modifying the Makefile, thanks for the heads up. I don't wish to make unwanted changes so I'll close this. It was also not very well thought out so it's probably for the best. Thank you for the reviews.

**Pull request closed without merging**